### PR TITLE
Feature/scan barcode to isbn

### PR DESCRIPTION
**Pull request by Huntsekkern**
No new tests written (I wrote one for the button and then deleted it) 
_As good UX requires to automatically move to the next activity on scanning, there's no button and it's impossible to write tests for intents without writing an AI which moves in Android Emulator VR. As none of my functions return something, it also seems tricky to write unit-tests. However, I tested manually extensively._

One known non-breaking issue:
Scanning a barcode might trigger several times the intent to start the next activity. Doesn't impede usage, but looks bad.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -45,13 +46,26 @@ dependencies {
...
+    implementation 'com.google.mlkit:barcode-scanning:16.1.1'
     testImplementation 'junit:junit:4.13.2'
     androidTestImplementation 'androidx.test.ext:junit:1.1.2'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.3.0'
     androidTestImplementation 'androidx.test:runner:1.3.0'
     androidTestImplementation 'androidx.test:rules:1.3.0'
     androidTestImplementation 'androidx.test.espresso:espresso-intents:3.3.0'
 
+    // CameraX core library using the camera2 implementation
+    def camerax_version = "1.0.0-rc03"
```
Comment (Huntsekkern) [inline app/build.gradle:58]: Cirrus says "A newer version of androidx.camera:camera-camera2 than 1.0.0-rc03 is available: 1.1.0-alpha02"
True, but I don't know whether we want to be on an alpha build, so it was a conscious decision.
Opinions?

File: app/src/androidTest/java/com/github/polybooks/AddSaleActivityTest.kt [hunk 2]
```diff
@@ -2,38 +2,45 @@ package com.github.polybooks
...
+
+    @Before
+    fun before() {
+        Intents.init()
+    }
+
+    @After
+    fun after() {
+        Intents.release()
+    }
```
Comment (Snowflake8) [inline app/src/androidTest/java/com/github/polybooks/AddSaleActivityTest.kt:32]: Ca fait quoi ça?

**Review by Snowflake8 — APPROVED** [review]
Comme tu l'as noté, la JavaDoc de tes classes (meme sommaire) manque

**Review by DGoedtkindtEPFL — APPROVED** [review]
The feature is not complete, but it seems like a good progression.

**Pull request merged**